### ISSUE TITLE
Support FP8 ProcessGroup in pytorch

### ISSUE THREAD

**Issue by tocean** [Azure/MS-AMP#50]
**What's the issue, what's expected?**:
GPT training need FP8-supported ProcessGroup to all reduce FP8 gradients in Data Parallism.
PyTorch ProcessGroup does not support FP8 data type.

**How to reproduce it?**:
Here is the ProcessGroupNccl definition in pytorch:
https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L53
Need to Add FP8-support like {at::kByte, ncclFp8E4M3}, {at::kChar, ncclFp8E5M2}

**Log message or shapshot?**:
![image](https://user-images.githubusercontent.com/4133758/232015379-f871add8-18e2-490e-97ee-cfc0413640d2.png)

**Additional information**:
We may work around it by compiling pytorch by-ourself and waiting the official support for fp8 in pytorch.
